### PR TITLE
[nuage_master] Adding module calls instead of command for idempotency.

### PR DESCRIPTION
**Pull request by kwoodson**

### CODE REVIEW

**Comment by kwoodson** [comment]
I'm unsure how to test these changes as I'm not familiar with nuage.  Please test or let me know how so I can test.  Thanks.

https://trello.com/c/ZPIwuaDr/373-2-idempotency-configure-role-user-permissions

**Comment by tbielawa** [comment]
testing now

**Comment by mtnbikenc** [comment]
Testing byo/config.yml with `openshift_use_nuage=true`

**Comment by kwoodson** [comment]
@vishpat, would you remind reviewing/testing these changes?

**Comment by kwoodson** [comment]
aos-ci-test

**Comment by mtnbikenc** [comment]
Testing nuage has turned out to be a bit more complicated.  I'll try to get this tested tomorrow.

**Comment by mtnbikenc** [comment]
@sdodson after reviewing the docs for Nuage, I'm not sure where to find RPMs for running the install.  Ideas on where I could find those, or is there a contact for someone more familiar with testing this component?

https://access.redhat.com/documentation/en-us/openshift_container_platform/3.5/html/installation_and_configuration/install-config-configuring-nuage-sdn

**Comment by sdodson** [comment]
@mtnbikenc @kwoodson -- @vishpat would know best. I'm fine with leaving this open until they're able to test.

File: roles/nuage_master/tasks/serviceaccount.yml [hunk 1]
```diff
@@ -1,42 +1,21 @@
...
     namespace: default
     state: present
   run_once: True
   delegate_to: "{{ nuage_ca_master }}"
 
 - name: Configure role/user permissions
-  command: >
-    {{ openshift.common.client_binary }} adm {{item}}
-    --config={{ nuage_tmp_conf }}
+  delegate_to: "{{ nuage_ca_master }}"
```
Comment (ingvagabund) [inline roles/nuage_master/tasks/serviceaccount.yml]: Maybe put this at the end of the `Configure role/user permissions` so it aligns with the remaining tasks.

File: roles/nuage_master/tasks/serviceaccount.yml [hunk 2]
```diff
@@ -1,42 +1,21 @@
...
+  oc_adm_policy_user:
+    namespace: default
+    resource_name: "{{ item.resource_name }}"
+    resource_kind: "{{ item.resource_kind }}"
+    user: "{{ item.user }}"
   with_items: "{{nuage_tasks}}"
```
Comment (ingvagabund) [inline roles/nuage_master/tasks/serviceaccount.yml]: s/`"{{nuage_tasks}}"`/`"{{ nuage_tasks }}"`

File: roles/nuage_master/vars/main.yaml [hunk 3]
```diff
@@ -23,4 +23,6 @@ nuage_master_crt_dir: /usr/share/nuage-openshift-monitor
 nuage_service_account: system:serviceaccount:default:nuage
 
 nuage_tasks:
-  - policy add-cluster-role-to-user cluster-reader {{ nuage_service_account }}
+- resource_kind: cluster-role
```
Comment (ingvagabund) [inline roles/nuage_master/vars/main.yaml]: Valid value based on [1]:
```
  resource_kind:
    description:
    - The kind of policy to affect
    required: true
    default: None
    choices: ["role", "cluster-role", "scc"]
    aliases: []
```

[1] https://github.com/openshift/openshift-ansible/blob/master/roles/lib_openshift/src/doc/policy_user

File: roles/nuage_master/tasks/serviceaccount.yml (hunk 1 above)
Comment (kwoodson) [inline roles/nuage_master/tasks/serviceaccount.yml]: I can move it if it is aesthetically pleasing.

File: roles/nuage_master/tasks/serviceaccount.yml (hunk 2 above)
Comment (kwoodson) [inline roles/nuage_master/tasks/serviceaccount.yml]: I can change this but was not in my original PR.

File: roles/nuage_master/tasks/serviceaccount.yml (hunk 1 above)
Comment (ingvagabund) [inline roles/nuage_master/tasks/serviceaccount.yml]: Not a blocker, just a nit.

**Comment by kwoodson** [comment]
@sdodson, what do you think about merging this so we don't lose the effort?  Do the tests run against nuage?

**Pull request merged**